src/Symfony/Component/Notifier/Message/SmsMessage.php

```diff
@@ -40,7 +41,15 @@ public static function fromNotification(Notification $notification, Recipient $r
             throw new LogicException(sprintf('To send a SMS message, "%s" should implement "%s" or the recipient should implement "%s".', get_debug_type($notification), SmsNotificationInterface::class, SmsRecipientInterface::class));
         }
 
-        return new self($recipient->getPhone(), $notification->getSubject());
+        $message = new self($recipient->getPhone(), $notification->getSubject());
+        $message->notification = $notification;
+
+        return $message;
+    }
+
+    public function getNotification(): ?Notification
```

Comment:
This should be reverted

Reply:
Hi @fabpot, you mean this function only or the entire modification of the file ?

Reply:
The whole change in the SmsMessage class.

Reply:
Question about that, I need to know the notification level inside the Moby Transport class to select a "type_quality" parameter (which implies cost differences). 

Is there an other way ?

Reply:
I would use the less costly by default and let's talk about this issue in another PR. That would allow to merge this one and think about the issue is a more broader sense instead of just for this specific provider.

Reply:
Thank for you feedback. Fix is on the way.